pylint/lint/pylinter.py

```diff
@@ -973,66 +976,64 @@ def check(self, files_or_modules):
                 files_or_modules,
             )
 
-    def check_single_file(self, name, filepath, modname):
-        """Check single file
+    def check_single_file(self, name: str, filepath: str, modname: str) -> None:
+        warnings.warn(
+            "In pylint 3.0, the checkers check_single_file function will only accept a FileItem",
```

Comment:
```suggestion
            "In pylint 3.0, the checkers check_single_file function will be removed. "
            "Use check_single_file_item instead.",
```
What do you think?

Reply:
Don't really have a preference. Guess this works as well and requires less refactoring for `3.0`.